api/tests/routes/backoffice/pro_test.py

```diff
@@ -769,6 +769,13 @@ def test_search_bank_account_by_iban(self, authenticated_client, search_query):
         finance_factories.BankAccountFactory(label="Other")
         self._search_for_one(authenticated_client, search_query, bank_account.id)
 
+    def test_search_bank_account_by_invoice_reference(self, authenticated_client):
+        bank_account = finance_factories.BankAccountFactory(label="Expected", iban="FR7612345000000123456789008")
+        finance_factories.InvoiceFactory(bankAccount=bank_account, reference="F250000550")
+        bank_account_2 = finance_factories.BankAccountFactory(label="Other")
+        finance_factories.InvoiceFactory(bankAccount=bank_account_2, reference="F250000551")
+        self._search_for_one(authenticated_client, "F250000550", bank_account.id)
```

Comment:
On peut toujours plus factoriser (si tu as envie, c'est vraiment pas nécessaire)
```suggestion
        expected_invoice = finance_factories.InvoiceFactory(bankAccount__label="Expected", reference="F250000550")
        finance_factories.InvoiceFactory(bankAccount__label="Other", reference="F250000551")
        self._search_for_one(authenticated_client, "F250000550", expected_invoice.bankAccountId)
```

Reply:
il n'y a effectivement pas besoins des labels 👍 , par conte, si j'utilise InvoiceFactory sans preciser de bankAccount, alors bankAccountId sera None, et aucun compte bancaire associé a l'invoice reference ne sera trouvé, ce qui fera fail le test

Reply:
Ah oui, le bank account est nullable tiens
Dans ce cas` InvoiceFactory(bankAccount=BankAccountFactory(), reference="F123432")` marchera

Reply:
je pourrai rajouter `bankAccount = factory.SubFactory(BankAccountFactory)` dans InvoiceFactory, mais le cela forcerai le liens vers un bankAccount, ce qui est contre intuitif avec le modele car le champs bankAccountId est nullable.

Reply:
j'ai d'ailleurs trouvé 208 invoice lié a aucun compte bancaire en staging 🤔